File: src/gazebo_change_start.py
#!/usr/bin/env python3

import rospy
from gazebo_msgs.srv import SetModelState
from gazebo_msgs.msg import ModelState

rospy.init_node('change_start', anonymous=True)
rospy.wait_for_service('/gazebo/set_model_state')

try:
    change_start = rospy.ServiceProxy('/gazebo/set_model_state', SetModelState)
    state = ModelState()
    state.model_name = 'turtlebot3_burger'
    state.pose.position.x = 1
    state.pose.position.y = 1
    state.pose.position.z = 0
    state.pose.orientation.x = 0
    state.pose.orientation.y = 0
    state.pose.orientation.z = 0
    state.pose.orientation.w = 1
    resp = change_start(state)
    print("Moved robot to x=1, y=1")
except rospy.ServiceException as e:
    print("Service call failed: %s"%e)

    

File: src/gazeboChangeStart.cpp
#include "ros/ros.h"
#include "gazebo_msgs/ModelState.h"
#include "gazebo_msgs/SetModelState.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "gazeboChangeStart");

    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<gazebo_msgs::SetModelState>("/gazebo/set_model_state");

    gazebo_msgs::SetModelState srv;
    srv.request.model_state.model_name = "turtlebot3_burger";
    srv.request.model_state.pose.position.x = 2;
    srv.request.model_state.pose.position.y = 1;
    srv.request.model_state.pose.position.z = 0;
    srv.request.model_state.pose.orientation.x = 0;
    srv.request.model_state.pose.orientation.y = 0;
    srv.request.model_state.pose.orientation.z = 0;
    srv.request.model_state.pose.orientation.w = 1;
    client.call(srv);

    return 0;
}
